Sources/UnleashProxyClientSwift/Metrics.swift

```diff
@@ -0,0 +1,151 @@
+import Foundation
+import SwiftEventBus
+
+extension Date {
+    func iso8601String() -> String {
+        let formatter = ISO8601DateFormatter()
+        return formatter.string(from: self)
+    }
+}
+
+struct ToggleMetrics {
+    var yes: Int = 0
+    var no: Int = 0
+    var variants: [String: Int] = [:]
+
+    func toJson() -> [String: Any] {
+        ["yes": yes, "no": no, "variants": variants]
+    }
+}
+
+struct Bucket {
+    private let clock: () -> Date
+    var start: Date
+    var stop: Date?
+    var toggles: [String: ToggleMetrics] = [:]
+
+    init(clock: @escaping () -> Date) {
+        self.clock = clock
+        start = clock()
+    }
+
+    mutating func closeBucket() {
+        stop = clock()
+    }
+
+    func isEmpty() -> Bool {
+        toggles.isEmpty
+    }
+
+    func toJson() -> [String: Any] {
+        let mappedToggles = toggles.mapValues { $0.toJson() }
+        return [
+            "start": start.iso8601String(),
+            "stop": stop?.iso8601String() ?? "",
+            "toggles": mappedToggles
+        ]
+    }
+}
+
+struct MetricsPayload {
+    let appName: String
+    let instanceId: String
+    let bucket: Bucket
+
+    func toJson() -> [String: Any] {
+        [
+            "appName": appName,
+            "instanceId": instanceId,
+            "bucket": bucket.toJson()
+        ]
+    }
+}
+
+public class Metrics {
+    let appName: String
+    let metricsInterval: TimeInterval
+    let clientKey: String
+    let poster: (URLRequest) async throws -> (Data, URLResponse)
+    let clock: () -> Date
+    var disableMetrics: Bool
+    var timer: Timer?
+    var bucket: Bucket
+    let url: URL
+
+    init(appName: String,
+         metricsInterval: TimeInterval,
+         clock: @escaping () -> Date,
+         disableMetrics: Bool = false,
+         poster: @escaping (URLRequest) async throws -> (Data, URLResponse),
+         url: URL,
+         clientKey: String) {
+        self.appName = appName
+        self.metricsInterval = metricsInterval
+        self.clock = clock
+        self.disableMetrics = disableMetrics
+        self.poster = poster
+        self.url = url
+        self.clientKey = clientKey
+        self.bucket = Bucket(clock: clock)
+    }
+
+    func start() {
+        guard !disableMetrics else { return }
```

Comment:
Would this be the same as 

```
if disableMetrics { return }
```

Don't know what's more idiomatic swift, but using an `if` seems more legible to me here 🤷🏼

Reply:
glad you raised it. I also though it's simpler but followed some of the guard idioms from swift. But I'll change it to your suggestion.